src/Data/Profunctor/Types.hs

```diff
@@ -243,3 +244,19 @@ instance Traversable (Forget r a) where
 instance Contravariant (Forget r a) where
   contramap _ (Forget k) = Forget k
   {-# INLINE contramap #-}
+
+-- | Via @Semigroup r => (a -> r)@
+-- @since 5.7.0
+instance Semigroup r => Semigroup (Forget r a b) where
+  Forget f <> Forget g = Forget (f <> g)
+  {-# INLINE (<>) #-}
+
+-- | Via @Monoid r => (a -> r)@
```

Comment:
Put a new line between this line and the `@since` annotation to ensure that Haddock renders the `@since` annotation on its own line.